### PR TITLE
[master] Fix compiler crashing issue for local var references in local record defs

### PR DESCRIPTION
**Pull request by pubudu91**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/resources/test-src/record/negative/local_var_ref_in_record.bal [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    record {|
+        string name;
+        int age = x;
+    |} person = {name: "John Doe"};
+
+    record {
+        string name;
+        int age = x;
+    } person = {name: "Jane Doe"};
```
Comment (hasithaa) [inline tests/jballerina-unit-test/src/test/resources/test-src/record/negative/local_var_ref_in_record.bal]: We need to assert person.age == x; afther this.

Comment (pubudu91) [reply tests/jballerina-unit-test/src/test/resources/test-src/record/negative/local_var_ref_in_record.bal]: This is a negative test

Comment (hasithaa) [reply tests/jballerina-unit-test/src/test/resources/test-src/record/negative/local_var_ref_in_record.bal]: Ahh, OK. But this is valid closure. I forgot we do not support these kinds of closures yet.

Comment (pubudu91) [reply tests/jballerina-unit-test/src/test/resources/test-src/record/negative/local_var_ref_in_record.bal]: Yeah, this should give a compile error in the current implementation

**Pull request merged**